crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -719,3 +720,85 @@ pub fn derive_from_world(input: TokenStream) -> TokenStream {
             }
     })
 }
+
+/// Automatically compose systems together with function syntax.
+///
+/// This macro provides some nice syntax on top of the `SystemRunner` `SystemParam`
+/// to allow running systems inside other systems. Overall, the macro accepts normal
+/// closure syntax:
+///
+/// ```ignore
+/// let system_a = |world: &mut World| { 10 };
+/// let system_b = |a: In<u32>, world: &mut World| { println!("{}", *a + 12) };
+/// compose! {
```

Comment:
Would it make sense to accept a function definition instead of or in addition to a closure?  That would make it easier for these systems to have names.  

For that matter, would it make sense to use an attribute macro, like

```rust
#[compose]
fn system_name() -> Result<(), RunSystemError> {
    run!(system_a)
}
```

?